cmd/ci-operator-prowgen/main.go

```diff
@@ -393,26 +411,54 @@ func main() {
 		os.Exit(0)
 	}
 
-	if len(opt.ciOperatorConfigPath) > 0 {
-		if jobConfig, _, _, err := generateProwJobsFromConfigFile(opt.ciOperatorConfigPath); err != nil {
-			fmt.Fprintf(os.Stderr, "%v\n", err)
+	if opt.fromReleaseRepo {
+		var err error
+		if opt.fromDir, err = getReleaseRepoDir("ci-operator/config"); err != nil {
+			fmt.Fprintf(os.Stderr, "--from-release-repo error: %v\n", err)
 			os.Exit(1)
-		} else {
-			if len(opt.prowJobConfigPath) > 0 {
-				err = mergeJobsIntoFile(opt.prowJobConfigPath, jobConfig)
-			} else {
-				err = writeJobs(jobConfig)
+		}
+	}
+	if opt.toReleaseRepo {
+		var err error
+		if opt.toDir, err = getReleaseRepoDir("ci-operator/jobs"); err != nil {
+			fmt.Fprintf(os.Stderr, "--to-release-repo error: %v\n", err)
+			os.Exit(1)
+		}
+	}
+
+	if len(opt.fromFile) > 0 {
+
+		jobConfig, org, repo, err := generateProwJobsFromConfigFile(opt.fromFile)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to generate jobs from '%s' (%v)\n", opt.fromFile, err)
+			os.Exit(1)
+		}
+		if len(opt.toFile) > 0 {
+			if err := mergeJobsIntoFile(opt.toFile, jobConfig); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to write jobs to '%s' (%v)\n", opt.toFile, err)
+				os.Exit(1)
 			}
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "failed to write the job configuration (%v)\n", err)
+		} else if len(opt.toDir) > 0 {
+			if err := writeJobsIntoComponentDirectory(opt.toDir, org, repo, jobConfig); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to write jobs to '%s' (%v)\n", opt.toDir, err)
 				os.Exit(1)
 			}
-
+		} else {
+			fmt.Fprintf(os.Stderr, "ci-operator-prowgen needs at least one of `--to-{file,dir,release-repo}` options\n")
```

Comment:
In a lot of our command-line tools we split up the input validation logic and execution logic so that when executing we can assume validity of input. Do you think that would potentially make this logic cleaner?

Reply:
Good point

Reply:
@stevekuznetsov PTAL now